alf/networks/networks.py

```diff
@@ -308,3 +315,186 @@ def forward(self, input, state):
                 lambda x: x.float() if x.dtype.is_floating_point else x, input)
         with torch.cuda.amp.autocast(self._enabled):
             return self._net(input, state)
+
+
+@alf.configurable
+@alf.repr_wrapper
+class NoisyFC(Network):
+    """The Noisy Linear Layer discribed in
+
+    Fortunato et. al. `Noisy Networks for Exploration <https://arxiv.org/abs/1706.10295>`_
+
+    Some details:
+
+    1. The noise for each sample in a batch is different.
+    2. The noise is maintained as state. It has a probability of `new_noise_prob`
+       to change to new noise.
```

Comment:
Without looking at the paper, it is not straightforward to see how the noise is applied.